to_be_ccppized/coords_1d.F90

```diff
@@ -1,152 +1,154 @@
+   ! Coordinates at cell mid-points.
+   real(r8), allocatable :: mid(:,:)
+   ! Width of cells.
+   real(r8), allocatable :: del(:,:)
+   ! Distance between cell midpoints.
+   real(r8), allocatable :: dst(:,:)
+   ! Reciprocals: 1/del and 1/dst.
+   real(r8), allocatable :: rdel(:,:)
+   real(r8), allocatable :: rdst(:,:)
+ contains
+   procedure :: section
+   procedure :: finalize            ! User-callable
+   final     :: finalize_Coords1D   ! Auto-finalize
+end type Coords1D
+
+interface Coords1D
+   module procedure new_Coords1D_from_fields
+   module procedure new_Coords1D_from_int
+end interface
+
+contains
+
+! Constructor to create an object from existing data.
+function new_Coords1D_from_fields(ifc, mid, del, dst, &
+     rdel, rdst) result(coords)
+  real(r8), USE_CONTIGUOUS intent(in) :: ifc(:,:)
+  real(r8), USE_CONTIGUOUS intent(in) :: mid(:,:)
+  real(r8), USE_CONTIGUOUS intent(in) :: del(:,:)
+  real(r8), USE_CONTIGUOUS intent(in) :: dst(:,:)
+  real(r8), USE_CONTIGUOUS intent(in) :: rdel(:,:)
+  real(r8), USE_CONTIGUOUS intent(in) :: rdst(:,:)
+  type(Coords1D) :: coords
+
+  coords = allocate_coords(size(ifc, 1), size(ifc, 2) - 1)
+
+  coords%ifc = ifc
+  coords%mid = mid
+  coords%del = del
+  coords%dst = dst
+  coords%rdel = rdel
+  coords%rdst = rdst
+
+end function new_Coords1D_from_fields
+
+! Constructor if you only have interface coordinates; derives all the other
+! fields.
+function new_Coords1D_from_int(ifc) result(coords)
+  real(r8), USE_CONTIGUOUS intent(in) :: ifc(:,:)
+  type(Coords1D) :: coords
+
+  coords = allocate_coords(size(ifc, 1), size(ifc, 2) - 1)
+
+  coords%ifc = ifc
+  coords%mid = 0.5_r8 * (ifc(:,:coords%d)+ifc(:,2:))
+  coords%del = coords%ifc(:,2:) - coords%ifc(:,:coords%d)
+  coords%dst = coords%mid(:,2:) - coords%mid(:,:coords%d-1)
+  coords%rdel = 1._r8/coords%del
+  coords%rdst = 1._r8/coords%dst
+
+end function new_Coords1D_from_int
+
+! Create a new Coords1D object that is a subsection of some other object,
+! e.g. if you want only the first m coordinates, use d_bnds=[1, m].
+!
+! Originally this used pointers, but it was found to actually be cheaper
+! in practice just to make a copy, especially since pointers can impede
+! optimization.
+function section(self, n_bnds, d_bnds)
+  class(Coords1D), intent(in) :: self
+  integer, intent(in) :: n_bnds(2), d_bnds(2)
+  type(Coords1D) :: section
+
+  section = allocate_coords(n_bnds(2)-n_bnds(1)+1, d_bnds(2)-d_bnds(1)+1)
+
+  ! The following code is intrepreted by nvhpc 24.9 as an allocation on assign
+  ! section%ifc  = self%ifc(n_bnds(1):n_bnds(2),d_bnds(1):d_bnds(2)+1)
+  ! section%mid  = self%mid(n_bnds(1):n_bnds(2),d_bnds(1):d_bnds(2))
+  ! section%del  = self%del(n_bnds(1):n_bnds(2),d_bnds(1):d_bnds(2))
+  ! section%dst  = self%dst(n_bnds(1):n_bnds(2),d_bnds(1):d_bnds(2)-1)
+  ! section%rdel = self%rdel(n_bnds(1):n_bnds(2),d_bnds(1):d_bnds(2))
+  ! section%rdst = self%rdst(n_bnds(1):n_bnds(2),d_bnds(1):d_bnds(2)-1)
```

Comment:
@johnmauff - Is this code something you wanted to have committed (typically we do not commit code which is commented out)

Reply:
I am fine with the elimination of the commented code.